test/memory.test.js

```diff
@@ -463,25 +463,88 @@ describe('Memory connector', function() {
       });
     });
 
+    it('automigrate all models - promise variant', function(done) {
+      ds.automigrate()
+        .then(function(result) {
+          done();
+        })
+        .catch(function(err){
+          done(err);
+        });
+    });
+
     it('automigrate one model', function(done) {
       ds.automigrate('m1', function(err) {
         done(err);
       });
     });
 
+    it('automigrate one model - promise variant', function(done) {
+      ds.automigrate('m1')
+        .then(function(result) {
+          done();
+        })
+        .catch(function(err){
+          done(err);
+        });
+    });
+
     it('automigrate one or more models in an array', function(done) {
       ds.automigrate(['m1'], function(err) {
         done(err);
       });
     });
 
+    it('automigrate one or more models in an array - promise variant', function(done) {
+      ds.automigrate(['m1'])
+        .then(function(result) {
+          done();
+        })
+        .catch(function(err){
+          done(err);
+        });
+    });
+
     it('automigrate reports errors for models not attached', function(done) {
       ds.automigrate(['m1', 'm2'], function(err) {
         err.should.be.an.instanceOf(Error);
         done();
       });
     });
+
+    it('automigrate reports errors for models not attached - promise variant', function(done) {
+      ds.automigrate(['m1', 'm2'])
+        .then(done)
```

Comment:
This will make the test pass even if there is no error reported. You need to provide a throwing `then` handler.

```js
.then(function() {
  done(new Error('automigrate() should have failed'));
})
```